andela/pippin-ah-backend
# [Chore: #163615731] Validate Input Types

#### What does this PR do?
- This PR checks for invalid input types and duplicate comments by a user in an article.

#### Description of Task to be completed?
- [x] add validation for input types
- [x] add validation to check if a user has already posted a particular comment in an article
- [x] write tests for changes

#### How should this be manually tested?
- Try registering a user with an invalid input type for the fields, like `object` or `array`
- You should get an error message with `invalid input type`
- Also, as a user, if you try to add an exact comment twice to an article, you should get an error message on the second try.

#### What are the relevant pivotal tracker stories?
[#163615731](https://www.pivotaltracker.com/story/show/163615731)
